examples/flow-counter/interfaces/react-redux.js

```diff
@@ -0,0 +1,12 @@
+import type { Dispatch } from 'redux';
+import type { Component } from 'react';
+
+declare module 'react-redux' {
+  declare type MapStateToProps<State, Props> = (state: State) => Props;
+  declare type MapDispatchToProps<Dispatchable, Props> = (dispatch: Dispatch<Dispatchable>) => Props;
+
+  declare function connect<State, Dispatchable, ComponentProps, DefaultProps, ComponentState, StateProps, DispatchProps: $Diff<ComponentProps, StateProps>>(
```

Comment:
Here I tried to tell flow that `DispatchProps + StateProps == ComponentProps`, using an internal feature I found here in Flow's React type declaration: https://github.com/facebook/flow/blob/master/lib/react.js#L131, but with no luck.